docs/cmake-presets.md

```diff
@@ -0,0 +1,272 @@
+
+> [!Note] Windows developers: CMake Tools selects the most recent version of `cl.exe` installed by default. You can specify a specific compiler version with the `toolset` option in `CMakePresets.json`. See [Configure Presets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html#configure-preset) and [Toolset Selection](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2016%202019.html#toolset-selection) for more information. If CMake and your generator are not installed with Visual Studio, then they will need to be on your `PATH` or in the default installation folder.
+
+## Add new Build Presets
+
+To add a new Test Preset to `CMakePresets.json`, run the **CMake: Add Test Preset** command. This lists several Test Preset templates in the command palette.
+* Select **Create from Configure Preset** to display a list of `configurePresets` defined in `CMakePresets.json`. Once you select a Configure Preset, an empty Test Preset associated with the selected Configure Preset will be created. 
+* Select **Inherit from Test Preset** to display a list of testPresets defined in `CMakePresets.json`. Once you select a Test Preset, a new Test Preset that inherits from the selected Test Preset will be created.
+* Select the **Custom** template to configure an empty Test Preset.
+For more information about editing Test Presets, see [Test Presets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html#test-preset).
+
+## Edit presets
+
+The official [CMake documentation](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html#id1) is the best resource for editing Configure Presets, Build Presets, and Test Presets. The following information is a subset of the CMake documentation that covers common actions.
+
+**Select your compilers**
+C and C++ compilers can be set with `cacheVariables.CMAKE_C_COMPILER` and `cacheVariables.CMAKE_CXX_COMPILER` in a Configure Preset. This is equivalent to passing `-D CMAKE_C_COMPILER=<value> and -D CMAKE_CXX_COMPILER=<value>` to CMake from the command line. For more information, see [CMAKE_<LANG>_COMPILER](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER.html#cmake-lang-compiler).
+
+You can specify the name of a compiler on your `PATH` or an environment variable that evaluates to the full path of a compiler. Full paths are discouraged so that the file will remain shareable.
+
+When you build with the Visual C++ toolset, CMake Tools automatically sources the environment from the latest version of the Visual Studio Build Tools installed on your system. You can specify a specific compiler version with the `toolset` option in `CMakePresets.json`. For more information, see [Configure Presets and Toolset Selection](https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html).
+
+A preset that builds for 64-bit Windows with cl.exe and a Visual Studio generator might look like this.
```

Comment:
Let's change this to: "A preset that builds for 64-bit Windows with cl.exe and a Visual Studio generator might set compilers like this."

Reply:
done